core/src/test/java/org/elasticsearch/index/percolator/ExtractQueryTermsServiceTests.java

```diff
@@ -218,6 +220,31 @@ public void testExtractQueryMetadata_boostQuery() {
         assertThat(terms.get(0).bytes(), equalTo(termQuery1.getTerm().bytes()));
     }
 
+    public void testExtractQueryMetadata_commonTermsQuery() {
```

Comment:
I like to prevent rest tests for these kind of things. Instead I think I can add a test for this query to the PercolatorQueryTests? Maybe like the already existing `testDuel()` test. We basically then test that the these queries yield the same result with and without query term extraction.